econtext/src/main/java/com/adyen/checkout/econtext/DefaultEContextDelegate.kt

```diff
@@ -0,0 +1,208 @@
+import com.adyen.checkout.components.ui.SubmitHandler
+import com.adyen.checkout.components.ui.Validation
+import com.adyen.checkout.components.ui.view.ButtonComponentViewType
+import com.adyen.checkout.components.ui.view.ComponentViewType
+import com.adyen.checkout.components.util.PaymentMethodTypes
+import com.adyen.checkout.components.util.ValidationUtils
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+
+@Suppress("TooManyFunctions", "LongParameterList")
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class DefaultEContextDelegate<EContextPaymentMethodT : EContextPaymentMethod>(
+    private val observerRepository: PaymentObserverRepository,
+    override val componentParams: ButtonComponentParams,
+    private val paymentMethod: PaymentMethod,
+    private val order: Order?,
+    private val analyticsRepository: AnalyticsRepository,
+    private val submitHandler: SubmitHandler<PaymentComponentState<EContextPaymentMethodT>>,
+    private val typedPaymentMethodFactory: () -> EContextPaymentMethodT,
+) : EContextDelegate<EContextPaymentMethodT> {
+
+    private val inputData: EContextInputData = EContextInputData()
+
+    private val _outputDataFlow = MutableStateFlow(createOutputData())
+    override val outputDataFlow: Flow<EContextOutputData> = _outputDataFlow
+
+    override val outputData: EContextOutputData get() = _outputDataFlow.value
+
+    private val _componentStateFlow = MutableStateFlow(createComponentState())
+    override val componentStateFlow: Flow<PaymentComponentState<EContextPaymentMethodT>> = _componentStateFlow
+
+    private val _viewFlow: MutableStateFlow<ComponentViewType?> = MutableStateFlow(EContextComponentViewType)
+    override val viewFlow: Flow<ComponentViewType?> = _viewFlow
+
+    override val submitFlow: Flow<PaymentComponentState<EContextPaymentMethodT>> = submitHandler.submitFlow
+    override val uiStateFlow: Flow<PaymentComponentUIState> = submitHandler.uiStateFlow
+    override val uiEventFlow: Flow<PaymentComponentUIEvent> = submitHandler.uiEventFlow
+
+    override fun initialize(coroutineScope: CoroutineScope) {
+        sendAnalyticsEvent(coroutineScope)
+    }
+
+    private fun sendAnalyticsEvent(coroutineScope: CoroutineScope) {
+        Logger.v(TAG, "sendAnalyticsEvent")
+        coroutineScope.launch {
+            analyticsRepository.sendAnalyticsEvent()
+        }
+    }
+
+    override fun updateInputData(update: EContextInputData.() -> Unit) {
+        inputData.update()
+        onInputDataChanged()
+    }
+
+    private fun onInputDataChanged() {
+        val outputData = createOutputData()
+        _outputDataFlow.tryEmit(outputData)
+        updateComponentState(outputData)
+    }
+
+    private fun createOutputData(): EContextOutputData {
+        return EContextOutputData(
```

Comment:
Minor: can we add names to call arguments since all of the are `FieldState<String>`?